### PR TITLE
Fixing Social User Refreshing

### PR DESCRIPTION
**Pull request by andrewconner**
- Using Try instead of Either
- Fixing Social user refreshing! Hurray!
- Adding logging to see what Karla is up to.

### CODE REVIEW

**Comment by andrewconner** [comment]
Additionally, `fetchAll()` was dead code.

File: shoebox/app/com/keepit/common/social/SocialGraphPlugin.scala [hunk 1]
```diff
@@ -92,15 +91,10 @@ class SocialGraphPluginImpl @Inject() (system: ActorSystem, socialGraph: Faceboo
...
-    val future = actor.ask(FetchAll)(1 minutes).mapTo[Int]
-    Await.result(future, 1 minutes)
-  }
-
-  override def asyncFetch(socialUserInfo: SocialUserInfo): Future[Either[Seq[SocialConnection], Exception]] = {
+  override def asyncFetch(socialUserInfo: SocialUserInfo): Future[Either[Exception, Seq[SocialConnection]]] = {
     require(socialUserInfo.credentials.isDefined,
       "social user info's credentials are not defined: %s".format(socialUserInfo))
-    actor.ask(FetchUserInfo(socialUserInfo))(5 minutes).mapTo[Either[Seq[SocialConnection], Exception]]
+    actor.ask(FetchUserInfo(socialUserInfo))(5 minutes).mapTo[Either[Exception, Seq[SocialConnection]]]
```
Comment (2is10) [inline shoebox/app/com/keepit/common/social/SocialGraphPlugin.scala]: Who dictates the "correct" ordering? If I were to choose a convention, I'd put the ordinary value first and the exception (or rare value) second.

**Comment by andrewconner** [comment]
@2is10 Mind quickly going over this? I want to push it to see what Karla Gonzales is up to :)

File: shoebox/app/com/keepit/controllers/ext/ExtEventController.scala [hunk 2]
```diff
@@ -41,7 +41,14 @@ class ExtEventController @Inject() (
 
   def logUserEvents = AuthenticatedJsonAction { request =>
     val userId = request.userId
-    val json = request.body.asJson.get
+
+    val json = try {
+      request.body.asJson.get
+    } catch {
+      case ex: java.util.NoSuchElementException =>
+        log.error(s"Bad event json payload from user id ${request.userId.id}\n${request.body}")
```
Comment (2is10) [inline shoebox/app/com/keepit/controllers/ext/ExtEventController.scala:49]: request headers might be helpful too

**Comment by 2is10** [comment]
Great, thanks.

**Pull request merged**